File: Week_09-高级动态规划/58.最后一个单词的长度.cpp
/*
 * @lc app=leetcode.cn id=58 lang=cpp
 *
 * [58] 最后一个单词的长度
 */

//从后往前，找到不是空格的单词，开始计数。

// @lc code=start
class Solution {
public:
    int lengthOfLastWord(string s) {
        if (s.size() < 1) return 0;
        int i = s.size()-1;
        int ans = 0;
        while (i >= 0 && s[i] == ' ') {
            i--;
        }
        while (i >= 0 && s[i] != ' ') {
            ans++;
            i--;
        }
        return ans;
    }
};
// @lc code=end

File: Week_09-高级动态规划/58.最后一个单词的长度.py
#
# @lc app=leetcode.cn id=58 lang=python
#
# [58] 最后一个单词的长度
#
# 倒序遍历字符串

# @lc code=start
class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        index = len(s) - 1
        while s[index] == ' ':
            index -= 1
        Len = 0
        while index >= 0 and s[index] != ' ':
            Len += 1
            index -= 1
        return Len
# @lc code=end

